pinot-plugins/pinot-input-format/pinot-protobuf/src/main/java/org/apache/pinot/plugin/inputformat/protobuf/ProtoBufUtils.java

```diff
@@ -48,21 +50,158 @@ public static InputStream getDescriptorFileInputStream(String descriptorFilePath
       PinotFS pinotFS = PinotFSFactory.create(scheme);
       Path localTmpDir = Files.createTempDirectory(TMP_DIR_PREFIX + System.currentTimeMillis());
       File protoDescriptorLocalFile = createLocalFile(descriptorFileURI, localTmpDir.toFile());
-      LOGGER.info("Copying protocol buffer descriptor file from source: {} to dst: {}", descriptorFilePath,
+      LOGGER.info("Copying protocol buffer jar/descriptor file from source: {} to dst: {}", filePath,
           protoDescriptorLocalFile.getAbsolutePath());
       pinotFS.copyToLocalFile(descriptorFileURI, protoDescriptorLocalFile);
-      return new FileInputStream(protoDescriptorLocalFile);
+      return protoDescriptorLocalFile;
```

Comment:
why this is named `protoDescriptorLocalFile`? i thought this can be jar file?

Reply:
It can be a jar of the descriptor file. It's used by both the codegen decoder (jar) and the dynamic decoder( descriptor)

Reply:
Renamed.